File: admin-web/src/api/operationLog.ts
import api from './index'

export interface OperationLogItem {
  id: number
  adminId: number | null
  adminName: string | null
  storeId: number | null
  operation: string
  method: string | null
  params: string | null
  ip: string | null
  status: number
  errorMsg: string | null
  createdAt: string
}

export interface OperationLogQuery {
  page?: number
  size?: number
  storeId?: number
  operation?: string
  status?: number
}

export const operationLogApi = {
  list: (params: OperationLogQuery) =>
    api.get<OperationLogItem[]>('/operation-log', { params }).then((r) => r.data),
}

File: backend/src/main/java/com/example/canteen/entity/OperationLog.java
package com.example.canteen.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 操作日志实体(对应 sys_operation_log 表)
 */
@Data
@TableName("sys_operation_log")
public class OperationLog {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long adminId;
    private String adminName;
    private Long storeId;
    private String operation;
    private String method;
    private String params;
    private String ip;
    /** 1=成功,0=失败 */
    private Integer status;
    private String errorMsg;
    private LocalDateTime createdAt;
}
